prow/github/client.go

```diff
@@ -3582,7 +3582,8 @@ func (c *client) EnsureFork(forkingUser, org, repo string) (string, error) {
 	if err != nil {
 		return repo, fmt.Errorf("could not fetch all existing repos: %v", err)
 	}
-	if !repoExists(fork, repos) {
+	// if the repo does not exist, or it does, but is not a fork of the repo we want
+	if forkedRepo := getFork(fork, repos); forkedRepo == nil || forkedRepo.Parent.FullName != fmt.Sprintf("%s/%s", org, repo) {
```

Comment:
In what scenario will the full name be wrong?

Reply:
The situation is we are bumping "istio/test-infra", and the robot's gh account has a repo called "test-infra" but that repo was forked from k8s/test-infra, so instead of just checking if "robot/test-infra" exists, we also check that it's parent's full name is "istio/test-infra", if not, the fork we want might not exist.

Reply:
Very clever, thanks!